CLIENTS.md

```diff
@@ -0,0 +1,631 @@
+        }
+    }
+}
+
+// DSSE entry
+{
+    "apiVersion": "0.0.2",
+    "kind": "dsse",
+    "spec": {
+        "dsseV0_0_2": {
+            "data": {
+                "algorithm": "SHA2_256",
+                // Note that we no longer include the envelope hash, since there is no
+                // canonicalization scheme for a DSSE envelope
+                "digest": "<base64-encoded digest of payload hash>"
+            },
+            "signatures": [
+                {
+                    "content": "<base64-encoded signature",
+                    "verifier": {
+                        // Must match signing algorithm
+                        "keyDetails": "PKIX_ECDSA_P256_SHA_256",
+                        "publicKey": {
+                            "rawBytes": "<base64 DER-encoded public key>"
+                        }
+                    }
+                }
+            ]
+        }
+    }
+}
+```
+
+### Two Entry Types
+
+Rekor v2 only supports `hashedrekord` (`HashedRekordLogEntryV0_0_2`) and
+`dsse` (`DSSELogEntryV0_0_2`) entry types, dropping a number of unused types
+such as `jar`, `alpine`, `rpm`, and the older types `rekord` and `intoto`.
+Additional types may be added in the future if there is demand, but this
+will require updating the client specification so that all clients implement
+support for these types.
+
+As with Rekor v1, the entry, aka `canonicalized_body`, will include the entry's
+kind (`hashedrekord`, `dsse`) and version (`0.0.2`) along with the entry itself
+in a `spec` field. Clients MUST gracefully fail when given a bundle with a
+kind or version that the client doesn't know how to parse. This is necessary
+so that clients gracefully fail when given a Rekor v2 entry.
+
+There are three possible scenarios for how clients should use kind and version
+to parse an entry:
+
+1. With a verification bundle with `canonicalized_body`, clients can use the
+   bundle's `kind` and `version` or `canonicalized_body.kind` and
+   `canonicalized_body.api_version` to parse `canonicalized_body.spec`.
+2. Clients that do not persist `canonicalized_body` must reconstruct the body
+   in order to compute the leaf hash to verify the inclusion proof. Clients can
+   use the bundle's `kind` and `version` to reconstruct the log entry given
+   the artifact. Remember to JSON-canonicalize the entry before computing the leaf hash.
+3. Monitors reading the entry from a tiled entry bundle will not have the verification
+   bundle, but will have the canonicalized body. A monitor must use the `kind` and
+   `apiVersion` to parse `spec`.
+
+### Certificate and Public Key Verifiers
+
+Rekor v2 only supports signature verification using a certificate or a
+public key, dropping support for PGP, minisign, pkcs7, SSH and TUF.
+Additional verifiers may be added in the future, but this will also require
+updating the client specification.
+
+### Handling Longer Requests
+
+Clients need to increase request timeouts to at least 10 seconds.
```

Comment:
```suggestion
Clients need to increase request timeouts to at least 10 seconds when creating entries.
```

Reply:
Added, and I've doubled the timeout recommendation to 20 seconds.